session/txn.go

```diff
@@ -217,14 +199,17 @@ func (txn *LazyTxn) GetOption(opt int) interface{} {
 func (txn *LazyTxn) changeInvalidToValid(kvTxn kv.Transaction) {
 	txn.Transaction = kvTxn
 	txn.initStmtBuf()
-	txn.recreateTxnInfo(
-		kvTxn.StartTS(),
-		txninfo.TxnRunningNormal,
-		uint64(txn.Transaction.Len()),
-		uint64(txn.Transaction.Size()),
-		"",
-		nil)
 	txn.txnFuture = nil
+
+	txn.mu.Lock()
+	defer txn.mu.Unlock()
+
```

Comment:
Can we write something like `txn.mu.TxnInfo = TxnInfo{}` to clean it up and avoid forgetting resetting any other field?

Reply:
It's there!

https://github.com/pingcap/tidb/pull/26040/files#diff-9dcf51b0e8ad2c9341d02bbf64b84753209d066a68b07091416a698f47c20bc5R259

Reply:
Emmm, I'm thinking you can call it every time the `recreateTxnInfo` is called in the old code, because it's expected to reset then unmentioned fields.